src/documents/ChangeApplierAPI.md

```diff
@@ -60,14 +60,90 @@ A `<modelPathReference>` has the form:
+
+#### The `path` entry in a model listener declaration ####
+
+The `path` entry holds basically the same function as the `<shortModelPathReference>` which may be held in the key of the listener declaration, but allows a richer set of path specifications to be used in order to
+specify which changes this model listener is interested in responding to. Note that if the `path` entry is supplied, then the key of the listener will be interpreted as a _namespace_ for the listener rather than as
+a path specification.
+
+<table>
+    <thead>
+        <tr>
+            <th colspan="3">Possible values for the <code>path</code> member of a model listener declaration</th>
+        </tr>
+        <tr>
+            <th>Type</th>
+            <th>Meaning</th>
+            <th>Examples</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td><code>String</code></td>
+            <td>A <code>&lt;shortModelPathReference&gt;</code></td>
+            <td>
+                See examples in section [Model Path References](#model-path-references)
+            </td>
+        </tr>
+        <tr>
+            <td><code>Object</code></td>
+            <td>A `<modelPathRecord>`, including members <ul>
+                <li><code>segs</code> (<code>Array</code> - required) and</li> 
+                <li><code>context</code> (<code>String</code> - optional) which encodes the model path to be matched.</li>
+                </ul>
+                </td>
+            <td>These examples encode exactly the same path expressions, in the same order as in the section [Model Path References](#model-path-references) 
+                <ul>
+                    <li><code>{segs: ["modelPath"]}</code></li>
+                    <li><code>{segs: ["modelPath", "&#42;"]}</code></li>
+                    <li><code>{segs: []}</code></li>
+                    <li><code>{segs: ["&#42;"]}</code></li>
+                    <li><code>{segs: ["modelPath"], context: "otherComponent"}</code></li>
+                    <li><code>{segs: ["modelPath", "&#42;"], context: "otherComponent"}</code></li>
+                    <li><code>{segs: [], context: "otherComponent"}</code></li>
+                    <li><code>{segs: ["&#42;"], context: "otherComponent"}</code></li>
+                </ul>
+            </td>
+        </tr>
+        <tr>
+            <td><code>Array</code></td>
+            <td>An array each of whose element may either be a String holding a <code>&lt;shortModelPathReference&gt;</code> or a <code>&lt;modelPathRecord&gt;</code>. The listener will be notified when <it>any</it> of these
+            paths receive changes. Below is further information on <a href="#matching-on-multiple-paths-in-a-single-model-listener-declaration"><code>matching on multiple paths</code></a></td>
+            <td><code>[
+                    "position", {
+                        segs: [["windowHolders"], "{that}.options.ourWindow"]
```

Comment:
Why is `"windowHolders"` in an array? Is the resolved model path different from the form without having it in an array - `segs: ["windowHolders", "{that}.options.ourWindow"]`?